pkg/yurthub/storage/disk/storage.go

```diff
@@ -46,408 +60,440 @@ func NewDiskStorage(dir string) (storage.Store, error) {
 		klog.Infof("disk cache path is empty, set it by default %s", CacheBaseDir)
 		dir = CacheBaseDir
 	}
-	if _, err := os.Stat(dir); os.IsNotExist(err) {
-		if err = os.MkdirAll(dir, 0755); err != nil {
-			return nil, err
-		}
+
+	fsOperator := &fs.FileSystemOperator{}
+
+	if err := fsOperator.CreateDir(dir); err != nil && err != fs.ErrExists {
+		return nil, fmt.Errorf("failed to create cache path %s, %v", dir, err)
 	}
 
+	// prune suffix "/" of dir
+	dir = strings.TrimSuffix(dir, "/")
+
 	ds := &diskStorage{
 		keyPendingStatus: make(map[string]struct{}),
 		baseDir:          dir,
+		serializer:       json.NewSerializerWithOptions(json.DefaultMetaFactory, scheme.Scheme, scheme.Scheme, json.SerializerOptions{}),
+		fsOperator:       fsOperator,
 	}
 
-	err := ds.Recover("")
+	ifEnhancement, err := ifEnhancement(ds.baseDir, *ds.fsOperator)
```

Comment:
how about rename var `ifEnhancement` to `enhancementMode`?

Reply:
fixed